model/src/main/kotlin/OrtResult.kt

```diff
@@ -403,10 +404,27 @@ data class OrtResult(
     fun getAdvisorResultsForId(id: Identifier): List<AdvisorResult> = advisorResultsById[id].orEmpty()
 
     /**
-     * Return all [RuleViolation]s contained in this [OrtResult].
+     * Return all [RuleViolation]s contained in this [OrtResult]. Optionally exclude resolved violations with
+     * [omitResolved] and remove violations below the [minSeverity].
      */
     @JsonIgnore
-    fun getRuleViolations(): List<RuleViolation> = evaluator?.violations.orEmpty()
+    fun getRuleViolations(omitResolved: Boolean = false, minSeverity: Severity? = null): List<RuleViolation> {
+        val allViolations = evaluator?.violations.orEmpty()
+
+        val violationsWithMinSeverity = allViolations.filter { minSeverity == null || it.severity >= minSeverity }
```

Comment:
Please avoid filter at all in case of `minSeverity == null` by doing something like

        val violationsWithMinSeverity = minSeverity?.let { allViolations.filter { it.severity >= minSeverity } }
            ?: allViolations

Reply:
Or maybe nicer:

        val violationsWithMinSeverity = when (minSeverity) {
            null -> allViolations
            else -> allViolations.filter { it.severity >= minSeverity }
        }